File: programmers/12905.cpp
// 가장 큰 정사각형 찾기
// https://programmers.co.kr/learn/courses/30/lessons/12905
/*
    브포는 됨
    dp써야함

#include <bits/stdc++.h>
using namespace std;
int n, m;
bool find(int y, int x, int range, vector<vector<int>> &board)
{
    for (int i = y; i < y + range; i++)
    {
        for (int j = x; j < x + range; j++)
        {
            if (board[i][j] == 0)
                return false;
        }
    }
    return true;
}
int solution(vector<vector<int>> board)
{
    int answer = 0;
    n = board.size();
    m = board[0].size();

    int len = min(n, m);

    while (len > 0)
    {
        bool check = true;
        for (int i = 0; i < n - (len - 1); i++)
        {
            for (int j = 0; j < m - (len - 1); j++)
            {
                if (find(i, j, len, board))
                {
                    answer = len * len;
                    return answer;
                }
            }
        }

        len--;
    }

    return answer;
}
*/

File: programmers/12905.py
def solution(board):

    for y in range(1, len(board)):
        for x in range(1, len(board[y])):
            if board[y][x] == 1:
                board[y][x] = min(
                    board[y-1][x-1], min(board[y-1][x], board[y][x-1])) + 1

    answer = 0

    for y in range(len(board)):
        for x in range(len(board[x])):
            answer = max(answer, board[y][x])
    return answer
    # return max([item for row in board for item in row])**2
